ykrt/try_passes.py

```diff
@@ -0,0 +1,190 @@
+#!/usr/bin/env python3
+
+import argparse
+import os, sys
+from subprocess import Popen, PIPE
+from dataclasses import dataclass
+import math
+import shutil
+from multiprocessing import Process, Manager, Pool, Lock
+import time
+
+# Stages in an LTO pipeline where optimisation passes can happen.
+STAGES = "pre_link", "link_time"
+CWD = None
```

Comment:
Whoah, global mutable variables! Except in very limited circumstances, we can't do this -- it will cause us all sorts of problems.